### ISSUE TITLE
registerAllFlagsListener passes a JSON string to its handler, not an array of flags

### ISSUE THREAD

**Issue by vcarl** [launchdarkly/react-native-client-sdk#93]
**Is this a support request?**
no

**Describe the bug**
An "all flags" listener is being passed a string representation of a JSON array, not the array itself (which is what the TS types specify it receives).

```js
const handler = (allFlags) => {
  console.log({ allFlags })
}
client.registerAllFlagsListener(handlerId, handler);
// assuming a flag called `'feature-flag'`, what's logged is:
// {"allFlags": "[\"feature-flag\"]"}
```

**To reproduce**
see above

**Expected behavior**
Argument values match their TS types

**Logs**
n/a

**SDK version**
4.0.3

**Language version, developer tools**
React native, I'm not sure what JS version the device is running

**OS/platform**
Android 11 simulator, Pixel 2 emulator

**Additional context**

**Comment by gwhelanLD** [comment]
This issue was fixed in the [5.0.1 release](https://github.com/launchdarkly/react-native-client-sdk/releases/tag/5.0.1) of the SDK. Is it possible for you to update to that version? If not, we can investigate back-porting that fix to the 4.0 SDK.

Thanks,
@gwhelanLD

**Comment by vcarl** [comment]
Should be able to update, thanks for the quick response. I'll close out.

While I'm here, the RN docs for "subscribing to updates" are extremely unhelpful. The code sample doesn't actually retrieve the update value, it just stores what flags have changed (in a super non-idiomatic manner), while variable names imply that it's the value itself being passed to the handler. 

```
if (this.state.listeners.hasOwnProperty(key)) {
  return;
}
let listener = value => Alert.alert('Listener Callback', value);
client.registerFeatureFlagListener('MY_LISTEN_FLAG_KEY', listener);
this.setState({listeners: {...this.state.listeners, ...{['MY_LISTEN_FLAG_KEY']: listener}}});
```
https://docs.launchdarkly.com/sdk/features/flag-changes#react-native

I settled on a solution that looks like this (with some abstractions collapsed), but I found it very complicated to actually retrieve the updated value.

```js
const experiments = {
  ["some-flag"]: {
    defaultValue: true,
  } /* an object of flag names/defaults */,
};

const newValues = await Promise.all(
  newFlags.map(async (flag) => {
    const { defaultValue } = experiments[flag];
    switch (typeof defaultValue) {
      case "string":
        return {
          name,
          value: await client.stringVariation(flag, defaultValue),
        };
      case "boolean":
        return {
          name,
          value: await client.boolVariation(flag, defaultValue),
        };
      case "number":
        return {
          name,
          value: await client.intVariation(flag, defaultValue),
        };
    }
  })
);
```
